indra/newview/llfloaterjoystick.cpp

```diff
@@ -78,15 +79,22 @@ BOOL CALLBACK di8_list_devices_callback(LPCDIDEVICEINSTANCE device_instance_ptr,
     // Capable of detecting devices like Oculus Rift
     if (device_instance_ptr && pvRef)
     {
+        // We always include the device if it is from 3DConnexion, otherwise
+        // if the GameControl feature is enabled then we perfer to route other devices to it.
         std::string product_name = utf16str_to_utf8str(llutf16string(device_instance_ptr->tszProductName));
-        S32 size = sizeof(GUID);
-        LLSD::Binary data; //just an std::vector
-        data.resize(size);
-        memcpy(&data[0], &device_instance_ptr->guidInstance /*POD _GUID*/, size);
-
-        LLFloaterJoystick * floater = (LLFloaterJoystick*)pvRef;
-        LLSD value = data;
-        floater->addDevice(product_name, value);
+        bool include_device = LLViewerJoystick::is3DConnexionDevice(product_name) || !LLGameControl::isEnabled();
```

Comment:
When `LLGameControl::isEnabled()` we  disable the generic game-controller support portion of the old LLViewerJoystick stuff, because otherwise there would be a feature conflict when trying to drive the avatar or camera using a game controller.